risc0/groth16/src/docker.rs

```diff
@@ -51,7 +51,7 @@ pub fn stark_to_snark(identity_p254_seal_bytes: &[u8]) -> Result<Seal> {
         .arg("--rm")
         .arg("-v")
         .arg(&format!("{}:/mnt", work_dir.to_string_lossy()))
-        .arg("risczero/risc0-groth16-prover:v2024-04-03.2")
+        .arg("risczero/risc0-groth16-prover:v2024-05-16")
```

Comment:
Generally we should add a suffix in case we need to create another image in the same day. (and so it's consistent)